File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let count = 0;
    let len = people.length;
    people.sort((a,b) => a-b);
    let start = 0;
    let end = len - 1;
    while (start<=end) {
        if (people[start] + people[end] <= limit) {
             start++;
            end--;
        } else {
            end--;
        }
        count++;
    }
    return count;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
# 몸무게가 제일 많이 나가는 사람과 제일 적게 나가는 사람을 매칭한다. 
# 둘의 합한 무게가 limit보다 작으면 범위를 줄여나간다.  

def solution(people, limit):
    count = 0 
    start = 0 
    end = len(people) - 1
    people.sort()
    
    while(start<=end):
        if (people[start] + people[end] <= limit):
            start += 1
            end -= 1
            count += 1
        else: 
            end -= 1
            count += 1
	
    return count